vault/core_test.go

```diff
@@ -186,6 +187,155 @@ func TestCore_Unseal_Single(t *testing.T) {
 	}
 }
 
+func TestCore_Unseal_DataVersion(t *testing.T) {
+	// Ignore version pre-release
+	version.VersionPrerelease = ""
+
+	// Case where local-version is greater than data-version
+	version.Version = "0.7.0"
+	c, keys, _ := TestCoreDataVersion(t, "0.6.0")
+
+	for _, key := range keys {
+		_, err := TestCoreUnseal(c, key)
```

Comment:
The shares and threshold are both set to 3, what would the break do in this case since it has to iterate through all the keys?